src/overlays/actors/ovl_En_Go/z_en_go.h

```diff
@@ -2,103 +2,142 @@
+ */
+typedef enum EnGoType {
+    /* 0 */ ENGO_F_0 = 0,
+    /* 1 */ ENGO_STRETCHER,        // Racers stretching before the race
+    /* 2 */ ENGO_SPECTATOR,        // Spectators to the race
+    /* 3 */ ENGO_GATEKEEPER,       // Gatekeeper to the Goron Shrine
+    /* 4 */ ENGO_GRAVEYARD,        // Gravemaker and other
+    /* 5 */ ENGO_ASIDE_STORE,      // Shrine Goron outside store
+    /* 6 */ ENGO_ASIDE_ELDERSROOM, // Shrine Goron outside Elder's room
+    /* 7 */ ENGO_ASIDE_ELDERSSON,  // Shrine Goron aside the Elder's Son
+    /* 8 */ ENGO_MEDIGORON,        // Medigoron selling Powder Kegs
+} EnGoType;
+
+/**
+ * Subtype for Gorons at the racetrack (ENGO_STRETCHER and ENGO_SPECTATOR)
+ */
+typedef enum EnGoRacetrackSubtype {
+    /* 0 */ ENGO_STRETCHER_A = 0,
+    /* 1 */ ENGO_STRETCHER_B,
+    /* 2 */ ENGO_STRETCHER_C,
+    /* 3 */ ENGO_STRETCHER_D,
+    /* 4 */ ENGO_STRETCHER_HAMSTRINGSIT,
+    /* 5 */ ENGO_STRETCHER_HAMSTRINGSTAND,
+    /* 6 */ ENGO_SPECTATOR_CHEER,
+    /* 7 */ ENGO_SPECTATOR_SHOUTING,
+} EnGoRacetrackSubtype;
+
+/**
+ * Subtype for Gorons outside the Goron Graveyard (ENGO_GRAVEYARD)
+ */
+typedef enum EnGoGraveyardSubtype {
+    /* 0 */ ENGO_GRAVEYARD_GRAVEMAKER = 0,
+    /* 1 */ ENGO_GRAVEYARD_FROZEN,
+} EnGoGraveyardSubtype;
 
 typedef struct EnGo {
     /* 0x000 */ Actor actor;
     /* 0x144 */ SkelAnime skelAnime;
     /* 0x188 */ EnGoActionFunc actionFunc;
-    /* 0x18C */ EnGoActionFunc unk_18C;
-    /* 0x190 */ EnGoActionFunc unk_190;
+    /* 0x18C */ EnGoActionFunc interruptedActionFunc;
+    /* 0x190 */ EnGoActionFunc graveyardDialogActionFunc;
     /* 0x194 */ ColliderCylinder colliderCylinder;
-    /* 0x1E0 */ UNK_TYPE1 unk1E0[0x4C];
+    /* 0x1E0 */ ColliderCylinder unusedCylinder;
     /* 0x22C */ ColliderSphere colliderSphere;
-    /* 0x284 */ Path* unk_284;
-    /* 0x288 */ s8 unk_288;
-    /* 0x289 */ s8 unk_289;
-    /* 0x28C */ s32 unk_28C;
-    /* 0x290 */ Vec3f unk_290;
-    /* 0x29C */ Vec3f unk_29C;
-    /* 0x2A8 */ Vec3s unk_2A8;
-    /* 0x2AE */ Vec3s unk_2AE;
-    /* 0x2B4 */ Vec3s jointTable[18];
-    /* 0x320 */ Vec3s morphTable[18];
-    /* 0x38C */ Actor* unk_38C;
-    /* 0x390 */ u16 unk_390;
-    /* 0x392 */ u16 unk_392;
-    /* 0x394 */ u8 cueId;
-    /* 0x398 */ f32 unk_398;
-    /* 0x39C */ f32 unk_39C;
-    /* 0x3A0 */ f32 unk_3A0;
-    /* 0x3A4 */ f32 unk_3A4;
-    /* 0x3A8 */ f32 unk_3A8;
+    /* 0x284 */ Path* gatekeeperPath;
+    /* 0x288 */ s8 objIndexTaisou;
+    /* 0x289 */ s8 objIndexHakuginDemo;
+    /* 0x28C */ s32 msgScriptResumePos;
+    /* 0x290 */ Vec3f headPos;
+    /* 0x29C */ Vec3f bodyPos;
+    /* 0x2A8 */ Vec3s headRot;
+    /* 0x2AE */ Vec3s bodyRot;
+    /* 0x2B4 */ Vec3s jointTable[GORON_LIMB_MAX];
+    /* 0x320 */ Vec3s morphTable[GORON_LIMB_MAX];
+    /* 0x38C */ Actor* attentionTarget;
+    /* 0x390 */ u16 actionFlags;
+    /* 0x392 */ u16 lastTextId;
+    /* 0x394 */ u8 springArrivalCueId;
+    /* 0x398 */ f32 currAnimPlaySpeed;
```

Comment:
```suggestion
    /* 0x398 */ f32 curAnimPlaySpeed;
```
We use 1 r instead of 2 to abbreviate `current`